File: Add Binary.js
var addBinary = function(a, b) {
    let carry = 0, result='';
    let ac = a.length-1, ab=b.length-1;
    while (ac>=0||ab>=0||carry){
        if (ac>=0){
            carry += parseInt(a[ac]);
            ac--;
        }
        if (ab>=0){
            carry += parseInt(b[ab]);
            ab--;
        }
        result = (carry % 2 ).toString() + result;
        carry = Math.floor(carry/2);
    }
    return result ;

};

File: Add Binary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        # defining the length of the input strings
        result = ""
        acount = len(a)-1
        bcount = len(b)-1
        carry = 0

        while acount>=0 or bcount>=0:
            totalSum = carry
            if acount>=0:
                totalSum += int(a[acount])
                acount -= 1
            if bcount>=0:
                totalSum += int(b[bcount])
                bcount -= 1
            result = str(totalSum%2) + result 
            carry = totalSum//2
        if carry > 0:
            result = str(1) + result
        return result 